cedar-drt/src/lean_impl.rs

```diff
@@ -197,30 +191,31 @@ impl LeanDefinitionalEngine {
         policies: &ast::PolicySet,
         entities: &Entities,
     ) -> TestResult<TestResponse> {
-        let req = Self::serialize_authorization_request(request, policies, entities);
-        let response = unsafe { isAuthorizedDRT(req) };
-        Self::deserialize_authorization_response(response)
-    }
-
-    fn serialize_evaluation_request(
-        request: &ast::Request,
-        entities: &Entities,
-        expr: &Expr,
-        expected: Option<&Expr>,
-    ) -> *mut lean_object {
-        let request: String = serde_json::to_string(&EvaluationRequest {
+        let request: String = serde_json::to_string(&AuthorizationRequest {
             request,
+            policies,
             entities,
-            expr,
-            expected,
         })
-        .expect("failed to serialize request, expression, or entities");
+        .expect("failed to serialize request, policies, or entities");
         let cstring = CString::new(request).expect("`CString::new` failed");
-        unsafe { lean_mk_string(cstring.as_ptr() as *const u8) }
+        // Lean with decrement the reference count when we pass this object: https://github.com/leanprover/lean4/blob/master/src/include/lean/lean.h
+        let req = unsafe { lean_mk_string(cstring.as_ptr() as *const u8) };
+        let response = unsafe { isAuthorizedDRT(req) };
+        // req can no longer be assumed to exist
+
+        let lean_obj_p = unsafe { lean_string_cstr(response) };
+        let lean_obj_cstr = unsafe { CStr::from_ptr(lean_obj_p as *const i8) };
+        let response_string = lean_obj_cstr
+            .to_str()
+            .expect("failed to convert Lean object to string")
+            .to_owned();
+        unsafe {
+            lean_dec(response);
+        };
```

Comment:
Is there a reason this code can't be in a `lean_obj_to_string` function like before? I feel like it'd be prettier to have it in one place (and will prevent me from forgetting `lean_dec` in the future 😉)